26_Autoexposure/main.cpp

```diff
@@ -0,0 +1,874 @@
+			auto cmdbuf = m_graphicsCmdBufs[0].get();
+			cmdbuf->reset(IGPUCommandBuffer::RESET_FLAGS::NONE);
+			auto ds = m_lumaPresentDS[1].get();
+
+			auto pc = AutoexposurePushData
+			{
+				.meteringWindowScaleX = MeteringWindowScale[0] * m_gpuImg->getCreationParameters().extent.width,
+				.meteringWindowScaleY = MeteringWindowScale[1] * m_gpuImg->getCreationParameters().extent.height,
+				.meteringWindowOffsetX = MeteringWindowOffset[0] * m_gpuImg->getCreationParameters().extent.width,
+				.meteringWindowOffsetY = MeteringWindowOffset[1] * m_gpuImg->getCreationParameters().extent.height,
+				.lumaMin = LumaMinMax[0],
+				.lumaMax = LumaMinMax[1],
+				.EV = m_EV,
+				.sampleCountX = SampleCount[0],
+				.sampleCountY = SampleCount[1],
+				.viewportSizeX = m_gpuImg->getCreationParameters().extent.width,
+				.viewportSizeY = m_gpuImg->getCreationParameters().extent.height,
+				.lumaMeterBDA = m_lumaGatherBDA
+			};
+
+			queue->startCapture();
+
+			cmdbuf->begin(IGPUCommandBuffer::USAGE::ONE_TIME_SUBMIT_BIT);
+
+			const VkRect2D currentRenderArea =
+			{
+				.offset = {0,0},
+				.extent = { m_window->getWidth(), m_window->getHeight() }
+			};
+			// set viewport
+			{
+				const asset::SViewport viewport =
+				{
+					.width = float(m_window->getWidth()),
+					.height = float(m_window->getHeight())
+				};
+				cmdbuf->setViewport({ &viewport, 1 });
+			}
+			cmdbuf->setScissor({ &currentRenderArea, 1 });
+
+			// begin the renderpass
+			{
+				const IGPUCommandBuffer::SClearColorValue clearValue = { .float32 = {1.f,0.f,1.f,1.f} };
+				auto scRes = static_cast<CDefaultSwapchainFramebuffers*>(m_surface->getSwapchainResources());
+				const IGPUCommandBuffer::SRenderpassBeginInfo info = {
+					.framebuffer = scRes->getFramebuffer(acquire.imageIndex),
+					.colorClearValues = &clearValue,
+					.depthStencilClearValues = nullptr,
+					.renderArea = currentRenderArea
+				};
+				cmdbuf->beginRenderPass(info, IGPUCommandBuffer::SUBPASS_CONTENTS::INLINE);
+			}
+
+			cmdbuf->bindGraphicsPipeline(m_presentPipeline.get());
+			cmdbuf->bindDescriptorSets(nbl::asset::EPBP_GRAPHICS, m_presentPipeline->getLayout(), 1, 1, &ds);
+			ext::FullScreenTriangle::recordDrawCall(cmdbuf);
+			cmdbuf->endRenderPass();
+
+			cmdbuf->end();
+
+			// submit
+			const IQueue::SSubmitInfo::SSemaphoreInfo rendered[1] = { {
+				.semaphore = m_presentSemaphore.get(),
+				.value = m_submitIx + 1,
+				// just as we've outputted all pixels, signal
+				.stageMask = PIPELINE_STAGE_FLAGS::COLOR_ATTACHMENT_OUTPUT_BIT
+			} };
+			{
+				const IQueue::SSubmitInfo::SCommandBufferInfo commandBuffers[1] = { {
+					.cmdbuf = cmdbuf
+				} };
+				// we don't need to wait for the transfer semaphore, because we submit everything to the same queue
+				const IQueue::SSubmitInfo::SSemaphoreInfo acquired[1] = { {
+					.semaphore = acquire.semaphore,
+					.value = acquire.acquireCount,
+					.stageMask = PIPELINE_STAGE_FLAGS::NONE
+				} };
```

Comment:
you need to wait for the compute queue semaphore (from the luma metering) as well as the acquire

because eventually you'll compute the EV in the tonemapping pipeline and stop doing any host wait on the compute queue.